ddtrace/internal/sampling.py

```diff
@@ -180,3 +189,65 @@ def apply_span_sampling_tags(self, span):
         span.set_metric(_SINGLE_SPAN_SAMPLING_RATE, self._sample_rate)
         if self._max_per_second:
             span.set_metric(_SINGLE_SPAN_SAMPLING_MAX_PER_SEC, self._max_per_second)
+
+
+def make_span_sampling_rules():
+    # type: () -> List[SpanSamplingRule]
+    json_rules = os.getenv("DD_SPAN_SAMPLING_RULES")
+    if not json_rules:
+        return []
+    else:
+        return _parse_rules(json_rules)
+
+
+def _parse_rules(rules):
+    sampling_rules = []
+    if rules is not None:
+        json_rules = []
+        try:
+            json_rules = json.loads(rules)
+            if json_rules is not list:
+                raise TypeError("DD_SPAN_SAMPLING_RULES is not list", json_rules)
+        except JSONDecodeError:
+            raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES={}".format(rules))
+        for rule in json_rules:
+            if rule is not dict:
+                raise TypeError("rule specified via DD_SPAN_SAMPLING_RULES is not a dictionary:{}".format(rule))
+            sample_rate = rule.get("sample_rate", 1.0)
+            service = rule.get("service")
+            name = rule.get("name")
+            max_per_second = rule.get("max_per_second")
+            if service is None and name is None:
+                raise SingleSpanSamplingError(
+                    "Neither service or name specified for single span sampling rule:{}".format(rule)
+                )
+            if (service and _unsupported_pattern(service)) or (name and _unsupported_pattern(name)):
+                raise UnsupportedGlobPatternError
+
+            try:
+                sampling_rule = SpanSamplingRule(
+                    sample_rate=sample_rate, service=service, name=name, max_per_second=max_per_second
```

Comment:
`sample_rate` here is `str|float` when it's typed in SpanSamplingRule to be a `float`

also `SpanSamplingRule` should accept only float for `sample_rate` (need to update this in a separate PR)

Reply:
similarly for the other types

Reply:
https://github.com/DataDog/dd-trace-py/pull/4023